Formula/topgrade.rb

```diff
@@ -0,0 +1,17 @@
+class Topgrade < Formula
+  desc "Upgrade all the things"
+  homepage "https://github.com/r-darwish/topgrade"
+  url "https://github.com/r-darwish/topgrade/archive/v0.11.0.tar.gz"
+  sha256 "2f005ddea8becff799514672f70a5331ed5b5edc9526efa85035314c4b27d208"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--root", prefix
+  end
+
+  test do
+    output = shell_output("#{bin}/topgrade --version")
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
This app has only one functionality, which is to update you system by invoking the upgrade step of every detected package manager, as described in the [readme](https://github.com/r-darwish/topgrade/blob/master/README.md).

I think it is a bad idea to actually run it as a part of the test, because it creates a test that alters the running system, so I had no choice but to perform the most basic sanity check, which is to task Topgrade to report its version.

Reply:
Any chance of a limited dry run, of sorts?

Reply:
@DomT4 I updated the formula to version 0.13.0 which contains the dry run mode. The test phase runs the dry-run mode and checks that it wants to upgrade Homebrew.